### ISSUE TITLE
Refreshing on search page causes page to 404 on certain conditions when it isn't supposed to

### ISSUE THREAD

**Issue by SleeplessOne1917** [LemmyNet/lemmy-ui#968]
### Issue Summary

There are certain conditions that can cause the search page to 404 in a situation the user doesn't expect it to.

### Steps to Reproduce

1. Go to the search page on a lemmy instance (tested on v0.17.2).
2. Select a type, listing type, sort, community, or creator
3. Your URL should look something like this: `http://localhost:1236/search/q//type/All/sort/TopAll/listing_type/Subscribed/community_id/0/creator_id/0/page/1`
4. **DO NOT** submit a text search
5. Refresh the browser
6. The page will show a 404

### Technical details

Using firefox on Windows 11.

### Suspected Cause

Looking at [routes.ts](https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/routes.ts#L143) I see a route that looks like this: `/search/q/:q/type/:type/sort/:sort/listing_type/:listing_type/community_id/:community_id/creator_id/:creator_id/page/:page`. I believe the issue is that refreshing the page when after using any of the sort/filter options without submitting a search causes the `q` part of the route to be an empty string, which causes inferno's router to not be able find the correct component for the route.

If you want my 2 cents, I think it would be better to have the various search options as part of the query string instead of part of the path. E.g., instead of the route looking like it does in the previous paragraph, it would be the following: `/search?q=<query>&type=<type>&sort=<sort>&listing_type=<listing_type>&community_id=<community_id>&creator_id=<creator_id>&page=<page>`.  [I found a good stack overflow on the topic that puts it better than I can about when to use path variables versus when to use query parameters](https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api):

> TL;DR: Best practice for RESTful API design is that path params are used to identify a specific resource or resources, while query parameters are used to sort/filter those resources.

Replace "resource" with "page" and it's basically the same idea. 

If interested, I could make a PR for it.

**Comment by dessalines** [comment]
I'd def be up for changing those path params to actual optional query params. I forget why I did it the wrong way, but I don't see any reason it wouldn't work. The isomorphic `fetchInitialData` on these routes might need to be reworked a little tho, as they are finding the correct option by doing a `.split('/')`

**Comment by SleeplessOne1917** [comment]
The PR for this is merged.
